QT5 Compilation issues on OS X

OS X 10.11.6 (El Capitan), Ostinato master (e1194a1), QT5 v5.10.1 installed by homebrew, Protobuf v3.5.1 installed by homebrew

This is the follow-up of issue https://github.com/pstavirs/ostinato/issues/257:

I build Ostinato master on my iMac by following .travis.yml:
```
brew install qt
brew link qt --force
brew install protobuf
export CPLUS_INCLUDE_PATH=/usr/local/include
export LIBRARY_PATH=/usr/local/lib
QT_SELECT=qt5 qmake
make
```

`make` builds the GUI and the drone without errors, but some warning, see [make.log.txt](https://github.com/pstavirs/ostinato/files/1996932/make.log.txt).

Starting the newly build GUI works fine, the connection to the port group gets green, but no interfaces are showing up. When using the official v0.9 drone, then the connection from the newly build GUI to the port group works, the interfaces are showing. I haven't made further testing with the GUI, but at first glance it looks fine.

When starting the newly build drone in terminal, then connecting to the drone by `telnet 127.0.0.1 7878`, then killing the drone with ^C, I get the following log: [drone_master.log.txt](https://github.com/pstavirs/ostinato/files/1996937/drone_master.log.txt). The log shows no signs of activity, when connecting to it. When using the GUI to connect, it also shows no log activity, here a wireshark trace:  [drone_connect.pcap.zip](https://github.com/pstavirs/ostinato/files/1996945/drone_connect.pcap.zip). It shows only a request from GUI to drone, but no answer.

As a comparison here the log of the original v0.9 drone, when starting it, then connecting by telnet: [drone_0.9.log.txt](https://github.com/pstavirs/ostinato/files/1996947/drone_0.9.log.txt).

So the issue is, that with my build the connection to the drone doesn't work. As the new GUI basically works with the official drone, the QT5 and Protobuf library should to be basically working.

My current idea is that this compilation warning might be the problem, as it warns about RpcServer::incomingConnection
```
cd rpc/ && ( test -e Makefile.pbrpc || /usr/local/bin/qmake -o Makefile.pbrpc /Users/behlers/GNS3/source/ostinato/rpc/pbrpc.pro ) && /Library/Developer/CommandLineTools/usr/bin/make -f Makefile.pbrpc 
/Library/Developer/CommandLineTools/usr/bin/clang++ -c -pipe -stdlib=libc++ -isystem /usr/local/Cellar/qt/5.10.1/include -O2 -fPIC -std=gnu++11  -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -mmacosx-version-min=10.10 -Wall -W -DHAVE_REMOTE -DQT_NO_DEBUG -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I/usr/local/Cellar/qt/5.10.1/lib/QtGui.framework/Headers -I/usr/local/Cellar/qt/5.10.1/lib/QtNetwork.framework/Headers -I/usr/local/Cellar/qt/5.10.1/lib/QtCore.framework/Headers -I. -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/OpenGL.framework/Headers -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/AGL.framework/Headers -I/usr/local/Cellar/qt/5.10.1/mkspecs/macx-clang -F/usr/local/Cellar/qt/5.10.1/lib -o rpcserver.o rpcserver.cpp
In file included from rpcserver.cpp:20:
./rpcserver.h:50:10: warning: 'RpcServer::incomingConnection' hides overloaded virtual function [-Woverloaded-virtual]
    void incomingConnection(int socketDescriptor);
         ^
/usr/local/Cellar/qt/5.10.1/lib/QtNetwork.framework/Headers/qtcpserver.h:94:18: note: hidden overloaded virtual function 'QTcpServer::incomingConnection' declared here: type mismatch at 1st parameter ('qintptr' (aka 'long long') vs 'int')
    virtual void incomingConnection(qintptr handle);
                 ^
1 warning generated.
```

But as I have no real experience with QT and Protobuf I'm lost here.

Here my changes to handle the warnings about DumpView::dataChanged and struct StatsTuple:

```
diff --git a/client/dumpview.cpp b/client/dumpview.cpp
index f895ea4..9ac5f7d 100644
--- a/client/dumpview.cpp
+++ b/client/dumpview.cpp
@@ -157,7 +157,8 @@ QRegion DumpView::visualRegionForSelection(

 //protected slots:
 void DumpView::dataChanged(const QModelIndex &/*topLeft*/,
-        const QModelIndex &/*bottomRight*/)
+        const QModelIndex &/*bottomRight*/,
+        const QVector<int> &/*roles*/)
 {
     // FIXME(HI)
     update();
diff --git a/client/dumpview.h b/client/dumpview.h
index b35a213..7a6c046 100644
--- a/client/dumpview.h
+++ b/client/dumpview.h
@@ -39,7 +39,8 @@ protected:
     QRegion visualRegionForSelection( const QItemSelection &selection ) const;
 protected slots:
     void dataChanged( const QModelIndex &topLeft,
-            const QModelIndex &bottomRight );
+            const QModelIndex &bottomRight,
+            const QVector<int> &roles );
     void selectionChanged( const QItemSelection &selected,
     const QItemSelection &deselected );
     void paintEvent(QPaintEvent *event);
diff --git a/server/pcaptxstats.h b/server/pcaptxstats.h
index ed134bb..99884e7 100644
--- a/server/pcaptxstats.h
+++ b/server/pcaptxstats.h
@@ -24,7 +24,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>

 #include <QThread>

-class StatsTuple;
+struct StatsTuple;

 class PcapTxStats : public QThread
 {
```

The remaining warnings are looking minor, so I don't  touch them. But that means, that Ostinato can't  be compiled with `qmake -config debug`, as then all warnings are treated as errors.

```
# grep warning: make.log
portmodel.cpp:121:19: warning: equality comparison with extraneous parentheses [-Wparentheses-equality]
portmodel.cpp:131:24: warning: equality comparison with extraneous parentheses [-Wparentheses-equality]
portmodel.cpp:173:19: warning: equality comparison with extraneous parentheses [-Wparentheses-equality]
portmodel.cpp:186:24: warning: equality comparison with extraneous parentheses [-Wparentheses-equality]
portmodel.cpp:190:24: warning: equality comparison with extraneous parentheses [-Wparentheses-equality]
streammodel.cpp:116:23: warning: equality comparison with extraneous parentheses [-Wparentheses-equality]
device.cpp:31:15: warning: unused variable 'kEthTypeIp4' [-Wunused-const-variable]
```

During my tests the QT5 version behaves quite well, I noticed only one glitch. In "Edit Stream / Protocol Data" the "Internet Protocol ver 4" part scrolls extremely choppy. I haven't noticed such a behavior in other parts of the GUI. And it doesn't happen in the original QT4 version. That definitely needs some attention.

![protocol_data](https://user-images.githubusercontent.com/896406/39967145-b8f2cc40-56b6-11e8-81cc-07ed2037b818.png)

--- a/client/dumpview.cpp
+++ b/client/dumpview.cpp
@@ -157,7 +157,8 @@ QRegion DumpView::visualRegionForSelection(
 
 //protected slots:
 void DumpView::dataChanged(const QModelIndex &/*topLeft*/, 
-        const QModelIndex &/*bottomRight*/)
+        const QModelIndex &/*bottomRight*/,
+        const QVector<int> &/*roles*/)
 {
     // FIXME(HI)
     update();

--- a/client/dumpview.h
+++ b/client/dumpview.h
@@ -39,7 +39,8 @@ protected:
     QRegion visualRegionForSelection( const QItemSelection &selection ) const;
 protected slots:
     void dataChanged( const QModelIndex &topLeft, 
-            const QModelIndex &bottomRight );
+            const QModelIndex &bottomRight,
+            const QVector<int> &roles );
     void selectionChanged( const QItemSelection &selected,
     const QItemSelection &deselected );
     void paintEvent(QPaintEvent *event);

--- a/rpc/rpcconn.cpp
+++ b/rpc/rpcconn.cpp
@@ -41,7 +41,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>
 
 static QThreadStorage<QString*> connId;
 
-RpcConnection::RpcConnection(int socketDescriptor, 
+RpcConnection::RpcConnection(qintptr socketDescriptor, 
                              ::google::protobuf::Service *service)
     : socketDescriptor(socketDescriptor),
       service(service)

--- a/rpc/rpcconn.h
+++ b/rpc/rpcconn.h
@@ -43,7 +43,7 @@ class RpcConnection : public QObject
     Q_OBJECT
 
 public:
-    RpcConnection(int socketDescriptor, ::google::protobuf::Service *service);
+    RpcConnection(qintptr socketDescriptor, ::google::protobuf::Service *service);
     virtual ~RpcConnection();
 
     static void connIdMsgHandler(QtMsgType type,
@@ -67,7 +67,7 @@ private slots:
     void on_clientSock_disconnected();
 
 private:
-    int socketDescriptor;
+    qintptr socketDescriptor;
     QTcpSocket *clientSock;
 
     ::google::protobuf::Service *service;

--- a/rpc/rpcserver.cpp
+++ b/rpc/rpcserver.cpp
@@ -62,7 +62,7 @@ bool RpcServer::registerService(::google::protobuf::Service *service,
     return true;
 }
 
-void RpcServer::incomingConnection(int socketDescriptor)
+void RpcServer::incomingConnection(qintptr socketDescriptor)
 {
     QThread *thread = new QThreadX; // FIXME:QThreadX pending Qt4.4+
     RpcConnection *conn = new RpcConnection(socketDescriptor, service);

--- a/rpc/rpcserver.h
+++ b/rpc/rpcserver.h
@@ -47,7 +47,7 @@ signals:
     void notifyClients(int notifType, SharedProtobufMessage notifData);
 
 protected:
-    void incomingConnection(int socketDescriptor);
+    void incomingConnection(qintptr socketDescriptor);
 
 private:
     ::google::protobuf::Service *service;

--- a/server/pcaptxstats.h
+++ b/server/pcaptxstats.h
@@ -24,7 +24,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>
 
 #include <QThread>
 
-class StatsTuple;
+struct StatsTuple;
 
 class PcapTxStats : public QThread
 {

